File: app/src/main/java/com/blog/demo/md/recyclerview/RecyclerViewLayoutSpanActivity.java
package com.blog.demo.md.recyclerview;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.blog.demo.R;

public class RecyclerViewLayoutSpanActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_material_design_recycler_view_normal);

        // 设置水平布局方式
        RecyclerView recyclerView = findViewById(R.id.recycler_view);

        GridLayoutManager gridSpanLayoutManager = new GridLayoutManager(this, 4);
        gridSpanLayoutManager.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
            @Override
            public int getSpanSize(int position) {
                if (position % 5 == 0) {
                    return 4;
                }
                return 1;
            }
        });
        recyclerView.setLayoutManager(gridSpanLayoutManager);
        recyclerView.setAdapter(new RecyclerViewGridAdapter(this));
    }

}

File: app/src/main/java/com/blog/demo/md/recyclerview/RecyclerViewLayoutSpanActivity.kt
package com.blog.demo.md.recyclerview

import android.app.Activity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.GridLayoutManager.SpanSizeLookup
import androidx.recyclerview.widget.RecyclerView
import com.blog.demo.R

class RecyclerViewLayoutSpanActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_material_design_recycler_view_normal)

        // 设置水平布局方式
        val recyclerView: RecyclerView = findViewById(R.id.recycler_view)
        val gridSpanLayoutManager = GridLayoutManager(this, 4)
        gridSpanLayoutManager.spanSizeLookup = object : SpanSizeLookup() {
            override fun getSpanSize(position: Int): Int {
                return if (position % 5 == 0) {
                    4
                } else {
                    1
                }
            }
        }
        recyclerView.layoutManager = gridSpanLayoutManager
        recyclerView.adapter = RecyclerViewGridAdapter(this)
    }

}